go/ephemeral/user_ek_test.go

```diff
@@ -57,3 +59,71 @@ func TestNewUserEK(t *testing.T) {
 }
 
 // TODO: test cases chat verify we can detect invalid signatures and bad metadata
+
+func TestDeviceRevokeNewUserEK(t *testing.T) {
+	tc := libkb.SetupTest(t, "kex2provision", 2)
+	defer tc.Cleanup()
+	NewEphemeralStorageAndInstall(tc.G)
+
+	// Include a paper key with this test user, so that we have something to
+	// revoke.
+	user, err := kbtest.CreateAndSignupFakeUserPaper("e", tc.G)
+	require.NoError(t, err)
+
+	// Confirm that the user has a userEK.
+	firstStatement, err := fetchUserEKStatement(context.Background(), tc.G)
+	require.EqualValues(t, firstStatement.CurrentUserEkMetadata.Generation, 1, "should start at userEK gen 1")
+
+	// Load the full user so that we can grab their devices.
+	arg := libkb.NewLoadUserByNameArg(tc.G, user.Username)
+	fullUser, err := libkb.LoadUser(arg)
+	if err != nil {
+		tc.T.Fatal(err)
+	}
+	paperKey := fullUser.GetComputedKeyInfos().PaperDevices()[0]
+
+	// Revoke the paper key.
+	revokeEngine := engine.NewRevokeDeviceEngine(engine.RevokeDeviceEngineArgs{ID: paperKey.ID}, tc.G)
+	ctx := &engine.Context{
+		LogUI:    tc.G.UI.GetLogUI(),
+		SecretUI: user.NewSecretUI(),
+	}
+	err = engine.RunEngine(revokeEngine, ctx)
+	require.NoError(t, err)
+
+	// Finally, confirm that the revocation above rolled a new userEK.
+	secondStatement, err := fetchUserEKStatement(context.Background(), tc.G)
+	require.EqualValues(t, secondStatement.CurrentUserEkMetadata.Generation, 2, "after revoke, should have userEK gen 2")
+	maxGen, err := tc.G.GetUserEKBoxStorage().MaxGeneration(context.Background())
```

Comment:
this doesn't actually call unbox, would be good to double check we can decrypt the key we made - i would remove this and verify that the fetched statement metadata is the same as what is locally stored:
```
userEK = userEkBoxStorage.Get(2)
assert no err && userEK.Metadata == secondStatement.CurrentUserEkMetadata
userEkBoxStorage.ClearCache()
userEkBoxStorage.Get(2) 
assert no err && userEK.Metadata == secondStatement.CurrentUserEkMetadata
```